optimum/neuron/models/qwen2/model.py

```diff
@@ -0,0 +1,291 @@
+# Copyright Amazon Web Services and its Affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import warnings
+
+import torch
+from transformers_neuronx import base, bucket, decoder, ops, utils
+from transformers_neuronx.config import NeuronConfig
+from transformers_neuronx.constants import KV_SHARD_PAD, LAYOUT_HSB
+from transformers_neuronx.llama.hlo import LlamaForSamplingNoEmbeddingHlo
+
+from .config import Qwen2Config
+from .modules import Qwen2ForCausalLM
+
+
+class Qwen2ForSampling(base.NeuronModelBase):
```

Comment:
what are the difference between Qwen's modeling and `LlamaForSampling`? Maybe mark it down by comments?